src/util.cpp

```diff
@@ -92,12 +92,28 @@ namespace rhost {
             return nc;
         }
 
+        std::wstring to_unicode(const char* buf, size_t len) {
+            // Convert 8-bit characters to Unicode via Windows CP. This guarantees
+            // if locale for non-Unicode programs is set correctly, user can type in
+            // their language. This does NOT guarantee that all languages can be used
+            // since R is not Unicode app. If host app is Unicode, it must perform
+            // checks if text being passed here is convertible to Unicode.
+            std::wstring ws;
+            size_t cch = strlen(buf);
+            if (cch > 0) {
+                ws.resize(cch);
+                RString2Unicode(&ws[0], (char*)buf, len);
```

Comment:
Actually, I see that there's already code to do UTF-to-wchar in `from_utf8` - it does it as the very first step. You can just refactor that out into a function.